### PR TITLE
ASR-15: Add Aggregator application starter

### PR DESCRIPTION
**Pull request by artembilan**
Fixes spring-cloud-stream-app-starters/app-starters-release#15

### CODE REVIEW

File: spring-cloud-starter-stream-aggregator/README.adoc [hunk 1]
```diff
@@ -0,0 +1,57 @@
+//tag::ref-doc[]
+= Aggregator Processor
+
+Use the `aggregator` application to combine batches of message to single one based on the correlation.
```
Comment (garyrussell) [inline spring-cloud-starter-stream-aggregator/README.adoc]: to combine multiple messages into one, based on some correlation mechanism.

File: spring-cloud-starter-stream-aggregator/README.adoc [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+This processor is fully based on the Aggregator component from http://docs.spring.io/spring-integration/reference/html/messaging-routing-chapter.html#aggregator[Spring Integration].
+So, please, consult there for use-cases and functionality.
+
+== Options
+
+The **$$aggregator$$** $$processor$$ has the following options:
+
+//tag::configuration-properties[]
+$$aggregator.correlation$$:: $$SpEL expression for correlation key$$ *($$String$$, default: `correlationId` header)*
+$$aggregator.release$$:: $$SpEL expression for release strategy$$ *($$String$$, default: `sequenceSize` header)*
```
Comment (garyrussell) [inline spring-cloud-starter-stream-aggregator/README.adoc]: default: group size == sequenceSize header.

File: spring-cloud-starter-stream-aggregator/README.adoc [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+$$aggregator.aggregation$$:: $$SpEL expression for aggregation strategy$$ *($$String$$, default: collection of payloads)*
+$$aggregator.groupTimeout$$:: $$SpEL expression for timeout to expiring uncompleted groups$$ *($$String$$, default: none)*
```
Comment (garyrussell) [inline spring-cloud-starter-stream-aggregator/README.adoc]: ...for a timeout to expire uncompleted...

File: spring-cloud-starter-stream-aggregator/README.adoc [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+$$aggregator.messageStoreType$$:: $$Message store type.
+Possible values are: simple, jdbc, redis, mongodb, gemfire $$ *($$String$$, default: `simple`)*
+$$aggregator.messageStoreEntity$$:: $$Persistence message store entity: table prefix in RDBMS, collection name in MongoDb, etc$$ *($$String$$, default: none)*
+//end::configuration-properties[]
+
+By default `aggregator` processor uses:
```
Comment (garyrussell) [inline spring-cloud-starter-stream-aggregator/README.adoc]: the aggregator

File: spring-cloud-starter-stream-aggregator/src/main/java/org/springframework/cloud/stream/app/aggregator/MessageStoreConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+import org.springframework.integration.mongodb.support.MongoDbMessageBytesConverter;
+import org.springframework.integration.redis.store.RedisMessageStore;
+import org.springframework.integration.store.MessageGroupStore;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.util.StringUtils;
+
+import com.gemstone.gemfire.cache.Region;
+
+/**
+ * @author Artem Bilan
```
Comment (garyrussell) [inline spring-cloud-starter-stream-aggregator/src/main/java/org/springframework/cloud/stream/app/aggregator/MessageStoreConfiguration.java]: javadocs

File: spring-cloud-starter-stream-aggregator/src/main/resources/META-INF/spring.factories [hunk 6]
```diff
@@ -0,0 +1,2 @@
+org.springframework.boot.env.EnvironmentPostProcessor=\
+  org.springframework.cloud.stream.app.aggregator.ExcludeStoresAutoConfigurationEnvironmentPostProcessor
```
Comment (garyrussell) [inline spring-cloud-starter-stream-aggregator/src/main/resources/META-INF/spring.factories]: newline

**Review by garyrussell — COMMENTED** [review]
Clever work!!

That's all - all minor - I can make the changes on merge if you agree.

**Comment by artembilan** [comment]
Will address myself soon.
Thank you!

**Comment by garyrussell** [comment]
`mvnw` is not executable - will fix on merge.

**Comment by garyrussell** [comment]
No `.mvn` either - fixing.

**Comment by artembilan** [comment]
Doh... I added it somehow to the `.gitignore`

**Comment by garyrussell** [comment]
Merged as d7d05541ecf89b03a279c16a2ac52c134907b218

**Pull request closed without merging**